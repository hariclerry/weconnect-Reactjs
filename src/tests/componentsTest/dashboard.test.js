import React from "react";
import {shallow, mount} from "enzyme";
import { MemoryRouter } from "react-router-dom";
import jwt from "jsonwebtoken";
import weconnectStore from "store";
import UserDashboard from "containers/business/dashboard";

const loginUserMock = { access_token: jwt.sign({ email: "clerry@gmail.com", password: "this199" }, "hard to guess string")}

describe('UserDashboard component',() =>{
  beforeEach(() => {
    localStorage.setItem("token", loginUserMock.access_token);

});
const wrapper = shallow(<UserDashboard email="clerry@gmail.com" businesses="Andela" username="hari"/>)
it('should match snapshot',() =>{
  expect(wrapper).toMatchSnapshot();
})

it("should render the UserDashboard component", () => {
  const mountComponent = shallow(<MemoryRouter><UserDashboard store = {weconnectStore}/></MemoryRouter>)
  expect(mountComponent.length).toBe(1);
  
});


})