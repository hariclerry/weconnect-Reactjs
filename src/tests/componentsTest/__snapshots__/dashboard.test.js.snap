// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDashboard component should match snapshot 1`] = `
<div>
  <_class
    options={Object {}}
  />
  <withRouter(Connect(NavBar)) />
  <div
    className="container"
  >
    <div>
      <h4>
        Hello 
        , welcome to your dashboard
      </h4>
    </div>
    <hr />
    <br />
    <div
      className="fixed-footer"
    >
      <div
        className="table-responsive"
      >
        <table
          className="table table-striped"
        >
          <thead>
            <td>
              <b>
                Name
              </b>
            </td>
            <td>
              <b>
                Category
              </b>
            </td>
            <td>
              <b>
                Location
              </b>
            </td>
            <td>
              <b>
                Description
              </b>
            </td>
            <td>
              <b>
                Review
              </b>
            </td>
          </thead>
          <tbody>
            <div>
               No Businesses found 
            </div>
          </tbody>
        </table>
      </div>
    </div>
  </div>
  <Footer />
</div>
`;
